src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx

```diff
@@ -23,35 +24,51 @@ import { Alert, Button, DrawStyle, LoadingPlaceholder, StackingMode, useStyles2
 import { reportAppInteraction, USER_EVENTS_ACTIONS, USER_EVENTS_PAGES } from 'services/analytics';
 import { getFilterBreakdownValueScene } from 'services/fields';
 import { getQueryRunner, setLeverColorOverrides } from 'services/panel';
-import { buildLokiQuery } from 'services/query';
+import { buildLokiQuery, renderLogQLStreamSelector } from 'services/query';
+import { getSortByPreference } from 'services/store';
 import {
   ALL_VARIABLE_VALUE,
-  LOG_STREAM_SELECTOR_EXPR,
-  VAR_FIELD_GROUP_BY,
+  LOG_EXPR_WITHOUT_STREAM_SELECTOR,
   VAR_FIELDS,
+  VAR_FIELD_GROUP_BY,
   VAR_LABELS,
 } from 'services/variables';
 import { ServiceScene } from '../ServiceScene';
+import { BreakdownSearchScene, getLabelValue } from './BreakdownSearchScene';
 import { ByFrameRepeater } from './ByFrameRepeater';
 import { FieldSelector } from './FieldSelector';
 import { LayoutSwitcher } from './LayoutSwitcher';
-import { StatusWrapper } from './StatusWrapper';
-import { BreakdownSearchScene, getLabelValue } from './BreakdownSearchScene';
 import { SortByScene, SortCriteriaChanged } from './SortByScene';
-import { getSortByPreference } from 'services/store';
+import { StatusWrapper } from './StatusWrapper';
+
+export class DetectedField {
```

Comment:
Apologies, I was pretty vague with my previous comment. Ideally, my suggestion would be to just move this to its own little module file. Can be under services, or feel free to introduce a `models` or `data` folder.

Reply:
Makes total sense!